src/wincng.c

```diff
@@ -1730,6 +1867,1212 @@ _libssh2_wincng_dsa_free(libssh2_dsa_ctx *dsa)
 #endif
 
 
+/*******************************************************************/
+/*
+ * Windows CNG backend: ECDSA helper functions
+ */
+
+#if LIBSSH2_ECDSA
+
+/*
+ * Decode an uncompressed point.
+ */
+static int
+_libssh2_wincng_ecdsa_decode_uncompressed_point(
+    IN const unsigned char *encoded_point,
+    IN size_t encoded_point_len,
+    OUT _libssh2_ecdsa_point *point)
+{
+    unsigned int curve;
+
+    if(!point) {
+        return LIBSSH2_ERROR_INVAL;
+    }
+
+    /* Verify that the point uses uncompressed format */
+    if(encoded_point_len == 0 || encoded_point[0] != 4) {
+        return LIBSSH2_ERROR_INVAL;
+    }
+
+    for(curve = 0; curve < ARRAY_SIZE(_wincng_ecdsa_algorithms); curve++) {
+        if(_wincng_ecdsa_algorithms[curve].point_length ==
+            (encoded_point_len - 1) / 2) {
+
+            point->curve = curve;
+
+            point->x = encoded_point + 1;
+            point->x_len = _wincng_ecdsa_algorithms[curve].point_length;
+
+            point->y = point->x + point->x_len;
+            point->y_len = _wincng_ecdsa_algorithms[curve].point_length;
+
+            return LIBSSH2_ERROR_NONE;
+        }
+    }
+
+    return LIBSSH2_ERROR_INVAL;
+}
+
+/*
+ * Create a IEEE P-1363 signature from a point.
+ *
+ * The IEEE P-1363 format is defined as r || s,
+ * where r and s are of the same length.
+ */
+static int
+_libssh2_wincng_p1363signature_from_point(IN const unsigned char *r,
+                                          IN size_t r_len,
+                                          IN const unsigned char *s,
+                                          IN size_t s_len,
+                                          IN libssh2_curve_type curve,
+                                          OUT PUCHAR *signature,
+                                          OUT size_t *signature_length)
+{
+    const unsigned char *r_trimmed;
+    const unsigned char *s_trimmed;
+    size_t r_trimmed_len;
+    size_t s_trimmed_len;
+
+    /* Validate parameters */
+    if(curve >= ARRAY_SIZE(_wincng_ecdsa_algorithms)) {
+        return LIBSSH2_ERROR_INVAL;
+    }
+
+    *signature = NULL;
+    *signature_length = (size_t)
+        _wincng_ecdsa_algorithms[curve].point_length * 2;
```

Comment:
Would it be useful to check for a possible overflow?

Reply:
`_wincng_ecdsa_algorithms[curve].point_length` is a constant, so I'd say that an overflow check is uneccessary.